### ISSUE TITLE
Support naming of tabs

### ISSUE THREAD

**Issue by zachlloyd** [warpdotdev/warp#83]
Issue title pretty much says it all - we would allow users to give tabs a custom name.

**Comment by ccpost** [comment]
I imagine you have some smoother UX in mind, but I've been using this alias forever to set iTerm's tab titles:
```bash
function title {
    echo -ne "\033]0;"$*"\007"
}
```

In Warp, it'll briefly flash to the manually-set title then go back to `bash`. Would be cool if the alias stuck, but also I'd happily take a command palette shortcut or similar. 😉

**Comment by elviskahoro** [comment]
From #456
@osi

The tab title format is `username@host:path`

For windows on my local machine, `username@host` is redundant; It is consuming space that would be (better) used by showing more of the path (as that's the larger difference for tabs on my machine.

Alternately, if the tabs were wider when there are few tabs, that would allow more of the path to be seen.

**Comment by elviskahoro** [comment]
@ccpost @mtmeyer @RealHarshThakur @Olson3R @mittalyashu @ntotten @willosbourne @mikeyk @clinejj @caseymatt

We're working on tab renaming now! If Warp had a dialog with common tab names, what options would you like to see included?

**Comment by RealHarshThakur** [comment]
I think suggestions (maybe based on process running) would be nice

**Comment by mtmeyer** [comment]
@elviskahoro I reckon I'd like to see the typical processes i find myself running, such as 'test' 'server' 'client'.  I spose anything you'd typically run a 'longer term' process for, for example `jest --watchAll` or a local server

**Comment by elliot-nelson** [comment]
Biggest request: remove `user@host:` prefix.

Even better, replace `path` with just `folder`.  (That is, if I'm in `~/dev/forks/rushstack`, I want the title bar to just show "rushstack".)  This way the tab titles are still useful even when I have 10 of them open.

**Comment by mutchco** [comment]
Agreed! I think I can almost replace tmux with your tabbing system, but I'd really like to be able to name tabs to different services.

**Comment by minhluuquang** [comment]
This is the function I miss from iterm2

**Comment by jonsterling** [comment]
I second the suggestion to just remove the `user@host:` part --- this fills up the entire tab width in my case, and so all my tabs look identical to each other. It is very rare for most people's workflow to have tabs that involve different users or different hosts, so it would be better to just show the path. I think this could be a higher priority than a more sophisticated approach to customizing or renaming tabs.

**Comment by zachabernathy** [comment]
I was about to open a suggestion to allow tab renaming, but alas, I have been beaten to it. I wholeheartedly second (third fourth) allow renaming and at the very least drop the `user@host:` part.

**Comment by noamr** [comment]
> I second the suggestion to just remove the `user@host:` part --- this fills up the entire tab width in my case, and so all my tabs look identical to each other. It is very rare for most people's workflow to have tabs that involve different users or different hosts, so it would be better to just show the path. I think this could be a higher priority than a more sophisticated approach to customizing or renaming tabs.

Yea this please

**Comment by nikitavoloboev** [comment]
+1 on this issue, it's very frustrating. Would love to see just the path of where the tab is located so `~/src` for example.

**Comment by XiaofengXie16** [comment]
Is there any estimated timeline for adding this feature to warp?

**Comment by ramykl** [comment]
+1 to this plus adding other customisation options at the table level like setting colours or theme. iTerm 2 allows this as well and it helps add a clear difference betwen tab eg: 
<img width="754" alt="Screen Shot 2022-05-11 at 1 31 16 PM" src="https://user-images.githubusercontent.com/5079228/167839655-c8f7f938-a0e7-4d29-b215-a33aa972cbb0.png">
s

**Comment by elviskahoro** [comment]
This feature launched. We can open up new issues for extra customizations like color.

**Comment by dannyneira** [comment]
> I imagine you have some smoother UX in mind, but I've been using this alias forever to set iTerm's tab titles:
> 
> ```shell
> function title {
>     echo -ne "\033]0;"$*"\007"
> }
> ```
> 
> In Warp, it'll briefly flash to the manually-set title then go back to `bash`. Would be cool if the alias stuck, but also I'd happily take a command palette shortcut or similar. 😉

Please see more on this here, in particular the Terminal Tip: https://docs.warp.dev/features/tabs#how-to-use-it
